File: majorProject/src/test/java/com/aocc/framework/ViewportTest.java
package com.aocc.framework;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class ViewportTest {

    @Test
    public void update_letterboxesWideScreenWithUniformScale() {
        Viewport viewport = new Viewport();
        viewport.update(2400, 1080);

        assertEquals(1080f / GameConstants.WORLD_HEIGHT, viewport.getScale(), 0.001f);
        assertTrue(viewport.getLetterboxDestRect().left > 0);
    }

    @Test
    public void screenToWorld_mapsCenterOfLetterboxToWorldCenter() {
        Viewport viewport = new Viewport();
        viewport.update(1920, 1080);

        int worldX = viewport.screenToWorldX(960);
        int worldY = viewport.screenToWorldY(540);

        assertEquals(GameConstants.WORLD_WIDTH / 2, worldX, 2);
        assertEquals(GameConstants.WORLD_HEIGHT / 2, worldY, 2);
    }
}

File: majorProject/src/test/kotlin/com/aocc/framework/ViewportTest.kt
package com.aocc.framework

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class ViewportTest {

    @Test
    fun update_letterboxesWideScreenWithUniformScale() {
        val viewport = Viewport()
        viewport.update(2400, 1080)

        assertEquals(1080f / GameConstants.WORLD_HEIGHT, viewport.getScale(), 0.001f)
        assertTrue(viewport.getLetterboxDestRect().left > 0)
    }

    @Test
    fun screenToWorld_mapsCenterOfLetterboxToWorldCenter() {
        val viewport = Viewport()
        viewport.update(1920, 1080)

        val worldX = viewport.screenToWorldX(960f)
        val worldY = viewport.screenToWorldY(540f)

        assertEquals(GameConstants.WORLD_WIDTH / 2f, worldX.toFloat(), 2f)
        assertEquals(GameConstants.WORLD_HEIGHT / 2f, worldY.toFloat(), 2f)
    }

    @Test
    fun offsets_centerLetterboxOnUltrawide() {
        val viewport = Viewport()
        viewport.update(2400, 1080)

        assertTrue(viewport.getOffsetX() > 0)
        assertEquals(0f, viewport.getOffsetY(), 0.001f)
    }
}
